### PR TITLE
Update reqwest to v0.9

### PR DESCRIPTION
**Pull request by inux39**
Older version of reqwest can't build by openssl crate error.

```
error: failed to run custom build command for `openssl v0.9.24`
```

### CODE REVIEW

**Comment by klausi** [comment]
Oh no, missed this PR and created a duplicate :-( in #44 

Discussion is in #43

File: src/lib.rs [hunk 1]
```diff
@@ -454,8 +455,9 @@ impl Mastodon {
 
             };
 
-            let mut headers = Headers::new();
-            headers.set(Authorization(Bearer { token: (*data.token).to_owned() }));
+            let mut headers = HeaderMap::new();
+            let auth = HeaderValue::from_str(&format!("Bearer {}", data.token));
+            headers.insert(reqwest::header::AUTHORIZATION, auth.unwrap());
```
Comment (Aaronepower) [inline src/lib.rs]: I generally prefer to not include the crate name if I have use statements at the top, can you replace this with `header::AUTHORIZATION`? You can replace the use statement at 67 with `use reqwest::{self, HeaderMap, HeaderValue};`.

File: src/lib.rs [hunk 2]
```diff
@@ -466,8 +468,9 @@ impl Mastodon {
 
     /// Creates a mastodon instance from the data struct.
     pub fn from_data(data: Data) -> Self {
-        let mut headers = Headers::new();
-        headers.set(Authorization(Bearer { token: (*data.token).to_owned() }));
+        let mut headers = HeaderMap::new();
+        let auth = HeaderValue::from_str(&format!("Bearer {}", data.token));
+        headers.insert(reqwest::header::AUTHORIZATION, auth.unwrap());
```
Comment (Aaronepower) [inline src/lib.rs]: Same as above.

File: src/page.rs [hunk 3]
```diff
@@ -96,19 +97,22 @@ fn get_links(response: &Response) -> Result<(Option<Url>, Option<Url>)> {
     let mut prev = None;
     let mut next = None;
 
-    if let Some(link_header) = response.headers().get::<Link>() {
-        for value in link_header.values() {
-            if let Some(relations) = value.rel() {
+    let link_header = response.headers().get_all(LINK);
+    for value in &link_header {
+        let val = value.to_str().unwrap();
+        let raw = std::convert::From::from(val);
```
Comment (Aaronepower) [inline src/page.rs]: Is there a reason you're calling `From::from` instead of `let raw = val.into();`?

File: src/page.rs [hunk 4]
```diff
@@ -96,19 +97,22 @@ fn get_links(response: &Response) -> Result<(Option<Url>, Option<Url>)> {
...
+        let parsed: Link = Header::parse_header(&raw).unwrap();
+        for link_value in parsed.values() {
```
Comment (Aaronepower) [inline src/page.rs]: Can you change this variable name back to `value`?

**Comment by inux39** [comment]
Thank you for your review!
Fixed those requests and error handling.

File: src/page.rs [hunk 5]
```diff
@@ -96,8 +97,11 @@ fn get_links(response: &Response) -> Result<(Option<Url>, Option<Url>)> {
     let mut prev = None;
     let mut next = None;
 
-    if let Some(link_header) = response.headers().get::<Link>() {
-        for value in link_header.values() {
+    let link_header = response.headers().get_all(LINK);
+    for value in &link_header {
+        let val = std::convert::From::from(value.to_str()?);
```
Comment (Aaronepower) [inline src/page.rs]: One last thing, can this be `let val = value.to_str()?.into();` instead?

**Comment by inux39** [comment]
I have to learn more about rust...

**Comment by ao2** [comment]
@inux39  maybe you can "rebase -i" and "push --force" the topic branch before it gets merged, to have a cleaner development history.

A commit message saying "instead into" can become confusing after some time has passed.

**Comment by Aaronepower** [comment]
@ao2 That isn't necessary as I squash and merge all pull requests made to my repositories so they'll be a single commit.

**Comment by ao2** [comment]
@Aaronepower in this case squashing the changes is fine, thank you, but please consider evaluating case by case if squashing is the best course of action. Sometimes it really makes sense to have multiple logically separate commits to address an issue.

Also, when you squash, do not blindly squash the commit messages; for instance the "fix requested changes" and "instead into" messages should not belong to the final commit message, and they can even be confusing to a reader.

Having said that, rest assured that your work on Mammut is appreciated, in any way you decide to do it. :smile: 

Ciao,
   Antonio

**Pull request merged**